internal/pkg/agent/application/upgrade/rollback_windows.go

```diff
@@ -19,11 +26,90 @@ const (
 	afterRestartDelay = 20 * time.Second
 )
 
-func invokeCmd(agentExecutable string) *exec.Cmd {
+func InvokeCmdWithArgs(executable string, args ...string) *exec.Cmd {
 	// #nosec G204 -- user cannot inject any parameters to this command
-	cmd := exec.Command(agentExecutable, watcherSubcommand,
-		"--path.config", paths.Config(),
-		"--path.home", paths.Top(),
-	)
+	cmd := exec.Command(executable, args...)
+
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		// Signals are sent to process groups, and child process are part of the
+		// parent's process group. So to send a signal to a
+		// child process and not have it also affect ourselves
+		// (the parent process), the child needs to be created in a new
+		// process group.
+		//
+		// Creating a child with CREATE_NEW_PROCESS_GROUP disables CTLR_C_EVENT
+		// handling for the child, so the only way to gracefully stop it is with
+		// a CTRL_BREAK_EVENT signal.
+		// https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags
+		//
+		// Watcher process will also need a console in order to receive CTRL_BREAK_EVENT on windows.
+		// Elastic Agent main process running as a service does not have a console allocated and the watcher process will also
+		// outlive its parent during an upgrade operation so we add the CREATE_NEW_CONSOLE flag.
+		CreationFlags: windows.CREATE_NEW_PROCESS_GROUP,
+	}
 	return cmd
 }
+
+func StartWatcherCmd(log *logger.Logger, createCmd cmdFactory, opts ...WatcherInvocationOpt) (*exec.Cmd, error) {
+
+	invocationOpts := applyWatcherInvocationOpts(opts...)
+
+	// allocConsole
+	r1, _, consoleErr := allocConsoleProc.Call()
```

Comment:
Is there any need to cleanup after this call to avoid resource leaks?

https://learn.microsoft.com/en-us/windows/console/freeconsole has a name that indicates it does this, but that's not really what it seems to do.

I can't find documentation suggesting we need to, but this has been a source of memory leaks in the past so I wanted to double check.